include/mraa_internal_types.h

```diff
@@ -490,6 +500,8 @@ typedef struct _board_t {
     mraa_adv_func_t* adv_func;    /**< Pointer to advanced function disptach table */
     struct _board_t* sub_platform;     /**< Pointer to sub platform */
     mraa_boolean_t chardev_capable;  /**< Decide what interface is being used: old sysfs or new char device*/
+    mraa_led_dev_t led_dev[MAX_LED_COUNT]; /**< Array of LED devices */
+    int led_dev_count; /**< Total onboard LED device count */
```

Comment:
I'd suggest to use `unsigned int` here as we're unlikely to have a negative number of those. Otherwise please correct the check at line 141 in `led.c` to account for possible negative values and not only 0.

Reply:
`unsigned int` makes sense here. Will change it